### ISSUE TITLE
Some already loaded feed lists are empty again after quiting and restarting

### ISSUE THREAD

**Issue by mglh** [newsboat/newsboat#923]
Steps  that produce the issue on my system:

1. Delete `cache.db`, if there is one
2. Create `urls` with the content shown below
3. Start newsboat
2. Load feed lists with `reload-all`
3. Quit  newsboat
4. Start newsboat again

Result:

**Some feed lists are emtpy again ("0/0").**

Newsboat version:

```
newsboat 2.19.0
System: Linux 4.9.0-12-amd64 (x86_64)
Compiler: g++ 5.5.0
ncurses: ncurses 6.2.20200212 (compiled with 6.2)
libcurl: libcurl/7.69.1 OpenSSL/1.1.1f zlib/1.2.11 (compiled with 7.69.1)
SQLite: 3.31.1 (compiled with 3.31.1)
libxml2: compiled with 2.9.10
```

System:

- Debian Stretch
- newsboat compiled using LinuxBrew

~/.newsboat/config:

```
### Misc

# Notification
notify-beep                  "no"
notify-program               "notify-send"
notify-format                "newsboat: finished reload \n%f unread feeds \n(%n unread articles total)"
notify-screen                "yes"
notify-xterm                 "yes"

#datetime-format              "%Y-%m-%d %H:%M:%S"
datetime-format              "%Y-%m-%d"

#articlelist-format           "%4i %f %D %6L %?T?;%-17T; ?%t"
#articlelist-format           "%4i %f %D %?T?;%-17T; ?%t"
#articlelist-format           "%4i %f %D %-60t" # Best yet
articlelist-format           "%4i %f %D %t"
#articlelist-format           "%4i %f %-70t"
#articlelist-format           "%4i %f %-50t"

pager                        "internal"
#always-display-description   "yes"

#text-width                   "80"

#html-renderer                "w3m -dump -T text/html | fmt -w 40"
#html-renderer                "elinks -dump | fmt -w 40"
#html-renderer                "lynx -dump -stdin | fmt -w 40"

bookmark-autopilot           "yes"
bookmark-interactive         "yes"
bookmark-cmd                 "~/.newsboat/newsboat_bookmark-cmd.sh"

max-browser-tabs 5

# Open with »o«
browser "xdg-open %u 2>/dev/null"

# Player for podcasts with podboat
#player "mpv"
#player "vlc"
player "audacious"

# Path where articles shall be saved to
# Note: Path / Directory has to exist (!)
save-path "/tmp/newsboat_saved_aricles"

# Download path for enclosures (podcasts)
# Path / Directory will be created if missing
#download-path "/home/USER/Downloads/%n/%F %H-%M-%S \# %t"
download-path "/tmp/podboat_downloads/%n/%F %H-%M-%S \# %t"



### Coloring


## Supported elements

# listnormal        a normal list item
# listfocus         the currently selected list item
# listnormal_unread an unread list item
# listfocus_unread  the currently selected and unread list item
# info              the info bars on top and bottom
# background        the application background
# article           the article text


## Supported colors

# black
# red
# green
# yellow
# blue
# magenta
# cyan
# white
# default
# color<n>, e.g. color123


## Supported attributes

# standout
# underline
# reverse
# blink
# dim
# bold
# protect
# invis

color listnormal        white black
color listnormal_unread yellow black
color listfocus         white blue
color listfocus_unread  yellow blue
color info              white red
color background        white black
color article           white black

#highlight feedlist "----- .* -----" white blue
highlight article "^Title: .*$" yellow red
highlight article "^Titel: .*$" yellow red
highlight article "^Link: http[s]\?:\/\/[^ ]*$" cyan black



### Key bindings


## Available dialogs:

# all (default)
# feedlist
# filebrowser
# help
# articlelist
# article
# tagselection
# filterselection
# urlview
# podbeuter
# dirbrowser


## Un-bindings

unbind-key "j"
unbind-key "k"

unbind-key "n"
unbind-key "p"

unbind-key "J"
unbind-key "K"

unbind-key "^N"
unbind-key "^P"

unbind-key "^K"

unbind-key "SPACE"
unbind-key "BACKSPACE"

unbind-key "o"
unbind-key "O"

unbind-key "l"

## Bindings

# feedlist

bind-key   "j"         next                   feedlist
bind-key   "k"         prev                   feedlist
bind-key   "SPACE"     pagedown               feedlist
bind-key   "BACKSPACE" pageup                 feedlist
bind-key   "n"         next-unread            feedlist
bind-key   "p"         prev-unread            feedlist
bind-key   "c"         toggle-show-read-feeds feedlist

# articlelist

bind-key   "j"         next                   articlelist
bind-key   "k"         prev                   articlelist
bind-key   "SPACE"     pagedown               articlelist
bind-key   "BACKSPACE" pageup                 articlelist
bind-key   "n"         next-unread            articlelist
bind-key   "p"         prev-unread            articlelist
bind-key   "c"         toggle-show-read-feeds articlelist
#bind-key   ""         next-feed              articlelist
#bind-key   ""         prev-feed              articlelist
#bind-key   ""         next-unread-feed       articlelist
#bind-key   ""         prev-unread-feed       articlelist

bind-key   "x"         open-in-browser        articlelist
#bind-key   "^S"        save-all               articlelist

# article

bind-key   "j"         next                   article
bind-key   "k"         prev                   article
bind-key   "n"         next-unread            article
bind-key   "p"         prev-unread            article
 
bind-key   "x"         open-in-browser        article


### Macros


# Open in elinks with <,l>
macro l set browser "lynx %u" ; open-in-browser ; set browser "xdg-open %u"

# Open in lynx with <,l>
macro e set browser "elinks %u" ; open-in-browser ; set browser "xdg-open %u"

# Open with mpv with <,v>
macro v set browser "mpv --geometry=50%% --title 'newsboat Video' %u" ; open-in-browser ; set browser "xdg-open %u"

# Open with mpv via youtube-dl with <,y>
macro y set browser "youtube-dl -o - '%u' | mpv --title='%u' --geometry=50%% --force-seekable=yes -" ; open-in-browser ; set browser "xdg-open %u"



### URL viewing


#macro u pipe-to "urlview"
#macro u pipe-to "urlscan"
```

~/.newsboat/urls:

```
"exec:curl --silent                   http://omegataupodcast.net/nurDeutsch.php" "~omega tau | Deutsche Folgen [MP3]"
"exec:curl --silent                   http://omegataupodcast.net/englishOnly.php" "~omega tau | Englische Folgen [MP3]"
"exec:curl --silent                   http://omegataupodcast.net/?feed=podcast" "~omega tau | Deutsche und englische Folgen #1 [MP3]"
"exec:curl --silent                   http://omegataupodcast.net/category/podcast/feed/" "~omega tau | Deutsche und englische Folgen #2 [MP3]"
"exec:curl --silent                   https://feeds.lagedernation.org/feeds/ldn-mp3.xml" "~Lage der Nation [MP3]"
"exec:curl --silent                   https://feeds.metaebene.me/lnp/mp3" "~Logbuch: Netzpolitik [MP3]"
"exec:curl --silent                   https://feeds.metaebene.me/lnp/opus" "~Logbuch: Netzpolitik [Opus]"
"exec:curl --silent                   http://feeds.soundcloud.com/users/soundcloud:users:284192316/sounds.rss" "~Stimmenfang [MP3]"
"exec:curl --silent                   https://feeds.metaebene.me/cre/mp3" "~CRE: Technik, Kultur, Gesellschaft [MP3]"
"exec:curl --silent                   https://feeds.metaebene.me/cre/opus" "~CRE: Technik, Kultur, Gesellschaft [Opus]"
"exec:curl --silent                   https://www.deutschlandfunknova.de/podcast/eine-stunde-talk" "~DLF Nova | Eine Stunde Talk [MP3]"
"exec:curl --silent                   https://www.deutschlandfunknova.de/podcast/einhundert" "~DLF Nova | Einhundert [MP3]"
"exec:curl --silent                   http://damals-tm-podcast.de/index.php/feed/damalstm-mp3/" "~Damals (tm) [MP3]"
"exec:curl --silent                   https://feeds.metaebene.me/freakshow/mp3" "~Freak Show [MP3]"
"exec:curl --silent                   https://feeds.metaebene.me/freakshow/opus" "~Freak Show [Opus]"
"exec:curl --silent                   https://ukw.fm/feed/mp3/" "~Unsere kleine Welt [MP3]"
"exec:curl --silent                   https://ukw.fm/feed/opus/" "~Unsere kleine Welt [Opus]"
"exec:curl --silent                   https://feeds.metaebene.me/nsfw/mp3" "~Not Safe For Work [MP3]    (2018-06-21)"
"exec:curl --silent                   https://feeds.metaebene.me/nsfw/opus" "~Not Safe For Work [Opus]   (2018-06-21)"
"exec:curl --silent                   https://feeds.metaebene.me/forschergeist/mp3" "~Forschergeist [MP3]"
"exec:curl --silent                   https://feeds.metaebene.me/forschergeist/opus" "~Forschergeist [Opus]"
"exec:curl --silent                   https://feeds.metaebene.me/raumzeit/mp3" "~Raumzeit [MP3]"
"exec:curl --silent                   https://feeds.metaebene.me/raumzeit/opus" "~Raumzeit [Opus]"
"exec:curl --silent                   https://feeds.metaebene.me/fokuseuropa/mp3" "~Fokus Europa [MP3]    (2019-11-28)"
"exec:curl --silent                   https://feeds.metaebene.me/fokuseuropa/opus" "~Fokus Europa [Opus]   (2019-11-28)"
"exec:curl --silent                   https://kanackischewelle.podigee.io/feed/mp3" "~Kanackische Welle [MP3]"
"exec:curl --silent                   https://www.psycho-talk.de/feed/mp3/" "~Psychotalk [MP3]"
"exec:curl --silent                   https://www.psycho-talk.de/feed/opus/" "~Psychotalk [Opus]"
"exec:curl --silent                   https://besser.demkontinuum.de/feed/mp3/" "~Besser -- Der Podcast [MP3]"
"exec:curl --silent                   https://derblankeschrott.podigee.io/feed/mp3" "~Der Blanke Schrott [MP3]"
"exec:curl --silent                   https://anyca.st/feed/mp3" "~ANYCAST [MP3]"
"exec:curl --silent                   https://feeds.fireside.fm/linuxunplugged/rss" "~LINUX Unplugged [MP3]"

```

**Comment by Minoru** [comment]
Hi! This sounds an awful lot like #748, but that bug was fixed in 2.19, so shouldn't affect you.

I don't see anything suspicious in your config file, but just in case, can you re-try with an empty one?

I don't have your ~/.newsboat/newsboat_link-clean.sh script. Can you re-try without it and without manual calls to `curl`?

Finally, does the bug always occur with the same feeds? It'd be nice to have one feed where it always reproduces and one feed where it doesn't, so we can compare the two and guess what differences might be causing this.

Thanks for the report!

**Comment by mglh** [comment]
Sorry, I overlooked the presence of `~/.newsboat/newsboat_link-clean.sh` in `urls`.
I have removed it in my post.

I have tried again without `~/.newsboat/newsboat_link-clean.sh` in `urls` and with an empty `config`: No change in the result.

These entries in `urls`are affected:

```
   4         (0/0) omega tau | Deutsche und englische Folgen #2 [MP3]
   7         (0/0) Logbuch: Netzpolitik [Opus]
  10         (0/0) CRE: Technik, Kultur, Gesellschaft [Opus]
  15         (0/0) Freak Show [Opus]
  17         (0/0) Unsere kleine Welt [Opus]
  19         (0/0) Not Safe For Work [Opus]   (2018-06-21)
  21         (0/0) Forschergeist [Opus]
  23         (0/0) Raumzeit [Opus]
  28         (0/0) Psychotalk [Opus]
```

In the meantime I noticed that the issue appears only when the feed lists are loaded all at once with `reload-all`. When a feed list (one of the aforementioned) is loaded seperately with `reload`, its content is still there after having quit and restarted.

Furthermore, it seems that the feeds which are affected do change when entries are deleted from or added to `urls`. But I have not yet found out whether there is a reproducible pattern here.

**Comment by Minoru** [comment]
Reproduced (with current `master`, b1884ff27f341c8d400432d50567ec4857374f0b)! This is a duplicate of https://github.com/akrennmair/newsbeuter/issues/193. In short: Opus and MP3 feeds use the same GUIDs for identical episodes, so Newsboat keeps just one copy, attached to one of the feeds.

We are still in the early days of figuring out how to fix this properly (#887).

Since Newsbeuter's tracker is archived now, I'll keep this issue open for further discussion.

**Comment by Minoru** [comment]
Oh, actually, we already have a place for discussion about this — #898. So let's close this one.
